esi_leap/api/controllers/v1/utils.py

```diff
@@ -161,12 +161,18 @@ def offer_get_dict_with_added_info(offer, project_list=None, node_list=None):
 
 def lease_get_dict_with_added_info(lease, project_list=None, node_list=None):
     resource = lease.resource_object()
+    empty_resource = 'Unknown'
 
     lease_dict = lease.to_dict()
     lease_dict['project'] = keystone.get_project_name(lease.project_id,
                                                       project_list)
     lease_dict['owner'] = keystone.get_project_name(lease.owner_id,
                                                     project_list)
-    lease_dict['resource'] = resource.get_resource_name(node_list)
+
+    if resource.get_resource_name(node_list):
+        lease_dict['resource'] = resource.get_resource_name(node_list)
+    else:
+        lease_dict['resource'] = empty_resource
```

Comment:
Thanks for adding the intermediate IronicNode - now if that returns properly, we shouldn't have to have this if statement anymore, since we'll actually have a node being returned; it's just one with empty values.

Reply:
Ah okay!